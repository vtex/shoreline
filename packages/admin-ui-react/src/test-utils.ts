import type { ReactElement } from 'react'
import type { RenderOptions } from '@testing-library/react'
import { render as baseRender } from '@testing-library/react'
import { createSystem } from './system'

const [ThemeProvider] = createSystem()

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

export { render }
