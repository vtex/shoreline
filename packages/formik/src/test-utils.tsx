import '@testing-library/jest-dom/extend-expect'
import React, { useRef } from 'react'
import { axe, toHaveNoViolations } from 'jest-axe'
import type { RenderOptions } from '@testing-library/react'
import {
  render as baseRender,
  waitFor,
  screen,
  fireEvent,
} from '@testing-library/react'
import { createSystem } from '@vtex/admin-ui-react'
import type { ReactElement } from 'react'
import { renderHook, act } from '@testing-library/react-hooks'

expect.extend(toHaveNoViolations)

const [ThemeProvider] = createSystem()

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

/**
 * HOC to wrap a component with its state hook
 * @example
 * // Cmp has the useCmp state hook
 * const StatefulCpm = withState(Cmp, () => useCmpState({ isSelected: false }))
 */
function withState(Component: any, useStateHook: () => any) {
  function StatefulComponent(props: any) {
    // Avoids an invalid hook call
    const ref = useRef(useStateHook())

    return <Component state={ref.current} {...props} />
  }

  StatefulComponent.displayName = Component.displayName

  return StatefulComponent
}

function jestMatchMedia() {
  /**
   * ðŸš§ Workaround for window.match media
   * @see https://jestjs.io/docs/en/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom
   */
  return Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // Deprecated
      removeListener: jest.fn(), // Deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  })
}

export {
  render,
  withState,
  axe,
  jestMatchMedia,
  renderHook,
  act,
  waitFor,
  screen,
  fireEvent,
}
