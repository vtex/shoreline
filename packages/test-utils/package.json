{
  "name": "@vtex/shoreline-test-utils",
  "version": "1.0.32",
  "description": "Shoreline test utilities",
  "main": "./dist/index.mjs",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.mts",
  "repository": "vtex/shoreline",
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org"
  },
  "files": [
    "dist"
  ],
  "exports": {
    ".": {
      "import": "./dist/index.mjs",
      "types": "./dist/index.d.mts"
    },
    "./setup": {
      "import": "./dist/setup.mjs",
      "types": "./dist/setup.d.mts"
    }
  },
  "engines": {
    "node": ">=20"
  },
  "scripts": {
    "prebuild": "rm -rf dist",
    "dev": "tsup --watch",
    "build": "npm run prebuild && tsup"
  },
  "peerDependencies": {
    "vitest": "0.x"
  },
  "devDependencies": {
    "vitest": "0.34.6"
  },
  "dependencies": {
    "@testing-library/dom": "9.3.4",
    "@testing-library/jest-dom": "6.4.6",
    "@testing-library/react": "14.3.1",
    "@testing-library/user-event": "14.5.2"
  }
}
