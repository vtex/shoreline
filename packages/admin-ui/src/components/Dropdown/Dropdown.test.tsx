import React from 'react'
import { render, axe } from '../../test-utils'
import type { UseSelectProps } from 'downshift'

import type { UseDropdownReturnValue } from './index'
import { Dropdown, useDropdownState } from './index'

interface DropdownStateProps<T> extends UseSelectProps<T> {
  children: (state: UseDropdownReturnValue<T>) => JSX.Element
}

function DropdownState<T>({ children, ...hookProps }: DropdownStateProps<T>) {
  const state = useDropdownState({ ...hookProps })

  return children(state)
}

describe('Dropdown tests', () => {
  beforeAll(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // Deprecated
        removeListener: jest.fn(), // Deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    })
  })

  it('should not have a11y violations', async () => {
    const { container } = render(
      <DropdownState
        items={['Yesterday', '7 days ago', '28 days ago', 'One year ago']}
        initialSelectedItem="7 days ago"
      >
        {(state) => (
          <Dropdown
            label="Date"
            items={['Yesterday', '7 days ago', '28 days ago', 'One year ago']}
            state={state}
          />
        )}
      </DropdownState>
    )

    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
