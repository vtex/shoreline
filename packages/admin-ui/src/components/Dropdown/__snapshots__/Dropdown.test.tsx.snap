// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown tests should match snapshot 1`] = `
<DocumentFragment>
  .emotion-5 {
  position: relative;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  margin: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
}

.emotion-1 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  visibility: visible;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.emotion-0 {
  width: 1.25rem;
  height: 1.25rem;
  min-height: 1.25rem;
  min-width: 1.25rem;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: unset;
  -webkit-box-align: unset;
  -ms-flex-align: unset;
  align-items: unset;
  -webkit-box-pack: unset;
  -ms-flex-pack: unset;
  -webkit-justify-content: unset;
  justify-content: unset;
  visibility: hidden;
  background: #FFFFFF;
  border: 1px solid hsla(0, 0%, 87%, 1);
  border-radius: 4px;
  box-shadow: 0rem 0rem 1rem 0rem rgba(0,0,0,0.10);
  cursor: pointer;
  outline: none;
  margin-top: 0.25rem;
  padding-top: 1rem;
  padding-bottom: 1rem;
  min-width: inherit;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  position: absolute;
  z-index: 999;
}

.emotion-4 >*:not(:last-child) {
  margin-bottom: 0.5rem;
}

.emotion-3 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  height: 40px;
  width: auto;
  padding-left: 0.75rem;
  padding-right: 0.5rem;
  color: #FFFFFF;
  background: hsla(222, 63%, 43%, 1);
}

.emotion-3:disabled {
  background: hsla(0, 0%, 90%, 1);
  color: hsla(0, 0%, 45%, 1);
}

.emotion-3:hover {
  color: #FFFFFF;
  background: hsla(222, 63%, 34%, 1);
}

.emotion-3:active {
  color: #FFFFFF;
  background: hsla(222, 63%, 25%, 1);
}

.emotion-3:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-3:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(222, 64%, 84%, 1);
}

.emotion-3 svg {
  margin: 0.25rem;
}

<div
    class="emotion-5"
  >
    <span
      style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 1px;"
    >
      <label
        for="downshift-3-toggle-button"
        id="downshift-3-label"
      >
        Species
      </label>
    </span>
    <button
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-labelledby="downshift-3-label downshift-3-toggle-button"
      class="emotion-3"
      id="downshift-3-toggle-button"
      type="button"
    >
      <div
        class="emotion-2"
      >
        <div
          class="emotion-1"
        >
          <svg
            class="emotion-0"
            fill="currentColor"
            height="1.25rem"
            viewBox="0 0 256 256"
            width="1.25rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="16rem"
              width="16rem"
            />
            <polyline
              fill="none"
              points="208 96 128 176 48 96"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="18"
            />
          </svg>
           Arabica
        </div>
      </div>
    </button>
    <div
      aria-labelledby="downshift-3-label"
      class="emotion-4"
      id="downshift-3-menu"
      role="listbox"
      tabindex="-1"
    />
  </div>
  .emotion-5 {
  position: relative;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  margin: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
}

.emotion-1 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  visibility: visible;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.emotion-0 {
  width: 1.25rem;
  height: 1.25rem;
  min-height: 1.25rem;
  min-width: 1.25rem;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: unset;
  -webkit-box-align: unset;
  -ms-flex-align: unset;
  align-items: unset;
  -webkit-box-pack: unset;
  -ms-flex-pack: unset;
  -webkit-justify-content: unset;
  justify-content: unset;
  visibility: hidden;
  background: #FFFFFF;
  border: 1px solid hsla(0, 0%, 87%, 1);
  border-radius: 4px;
  box-shadow: 0rem 0rem 1rem 0rem rgba(0,0,0,0.10);
  cursor: pointer;
  outline: none;
  margin-top: 0.25rem;
  padding-top: 1rem;
  padding-bottom: 1rem;
  min-width: inherit;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  position: absolute;
  z-index: 999;
}

.emotion-4 >*:not(:last-child) {
  margin-bottom: 0.5rem;
}

.emotion-3 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  height: 40px;
  width: auto;
  padding-left: 0.75rem;
  padding-right: 0.5rem;
  color: hsla(222, 63%, 34%, 1);
  background: hsla(222, 62%, 92%, 1);
}

.emotion-3:disabled {
  background: hsla(0, 0%, 90%, 1);
  color: hsla(0, 0%, 45%, 1);
}

.emotion-3:hover {
  color: hsla(222, 63%, 34%, 1);
  background: hsla(222, 64%, 84%, 1);
}

.emotion-3:active {
  color: hsla(222, 63%, 34%, 1);
  background: hsla(222, 78%, 77%, 1);
}

.emotion-3:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-3:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(222, 64%, 84%, 1);
}

.emotion-3 svg {
  margin: 0.25rem;
}

<div
    class="emotion-5"
  >
    <span
      style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 1px;"
    >
      <label
        for="downshift-4-toggle-button"
        id="downshift-4-label"
      >
        Brew Methods
      </label>
    </span>
    <button
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-labelledby="downshift-4-label downshift-4-toggle-button"
      class="emotion-3"
      id="downshift-4-toggle-button"
      type="button"
    >
      <div
        class="emotion-2"
      >
        <div
          class="emotion-1"
        >
          <svg
            class="emotion-0"
            fill="currentColor"
            height="1.25rem"
            viewBox="0 0 256 256"
            width="1.25rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="16rem"
              width="16rem"
            />
            <polyline
              fill="none"
              points="208 96 128 176 48 96"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="18"
            />
          </svg>
           French Press
        </div>
      </div>
    </button>
    <div
      aria-labelledby="downshift-4-label"
      class="emotion-4"
      id="downshift-4-menu"
      role="listbox"
      tabindex="-1"
    />
  </div>
  .emotion-5 {
  position: relative;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  margin: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
}

.emotion-1 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  visibility: visible;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.emotion-0 {
  width: 1.25rem;
  height: 1.25rem;
  min-height: 1.25rem;
  min-width: 1.25rem;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: unset;
  -webkit-box-align: unset;
  -ms-flex-align: unset;
  align-items: unset;
  -webkit-box-pack: unset;
  -ms-flex-pack: unset;
  -webkit-justify-content: unset;
  justify-content: unset;
  visibility: hidden;
  background: #FFFFFF;
  border: 1px solid hsla(0, 0%, 87%, 1);
  border-radius: 4px;
  box-shadow: 0rem 0rem 1rem 0rem rgba(0,0,0,0.10);
  cursor: pointer;
  outline: none;
  margin-top: 0.25rem;
  padding-top: 1rem;
  padding-bottom: 1rem;
  min-width: inherit;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  position: absolute;
  z-index: 999;
}

.emotion-4 >*:not(:last-child) {
  margin-bottom: 0.5rem;
}

.emotion-3 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  height: 40px;
  width: auto;
  padding-left: 0.75rem;
  padding-right: 0.5rem;
  color: hsla(222, 63%, 43%, 1);
  background: transparent;
}

.emotion-3:disabled {
  background: hsla(0, 0%, 90%, 1);
  color: hsla(0, 0%, 45%, 1);
}

.emotion-3:hover {
  color: hsla(222, 63%, 43%, 1);
  background: hsla(224, 65%, 97%, 1);
}

.emotion-3:active {
  color: hsla(222, 63%, 43%, 1);
  background: hsla(222, 62%, 92%, 1);
}

.emotion-3:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-3:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(222, 64%, 84%, 1);
}

.emotion-3 svg {
  margin: 0.25rem;
}

<div
    class="emotion-5"
  >
    <span
      style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 1px;"
    >
      <label
        for="downshift-5-toggle-button"
        id="downshift-5-label"
      >
        Recipes
      </label>
    </span>
    <button
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-labelledby="downshift-5-label downshift-5-toggle-button"
      class="emotion-3"
      id="downshift-5-toggle-button"
      type="button"
    >
      <div
        class="emotion-2"
      >
        <div
          class="emotion-1"
        >
          <svg
            class="emotion-0"
            fill="currentColor"
            height="1.25rem"
            viewBox="0 0 256 256"
            width="1.25rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="16rem"
              width="16rem"
            />
            <polyline
              fill="none"
              points="208 96 128 176 48 96"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="18"
            />
          </svg>
           Latte
        </div>
      </div>
    </button>
    <div
      aria-labelledby="downshift-5-label"
      class="emotion-4"
      id="downshift-5-menu"
      role="listbox"
      tabindex="-1"
    />
  </div>
</DocumentFragment>
`;

exports[`Dropdown tests should render a Dropdown with a list of string and other with a list of object 1`] = `
<DocumentFragment>
  .emotion-5 {
  position: relative;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  margin: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
}

.emotion-1 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  visibility: visible;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.emotion-0 {
  width: 1.25rem;
  height: 1.25rem;
  min-height: 1.25rem;
  min-width: 1.25rem;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: unset;
  -webkit-box-align: unset;
  -ms-flex-align: unset;
  align-items: unset;
  -webkit-box-pack: unset;
  -ms-flex-pack: unset;
  -webkit-justify-content: unset;
  justify-content: unset;
  visibility: hidden;
  background: #FFFFFF;
  border: 1px solid hsla(0, 0%, 87%, 1);
  border-radius: 4px;
  box-shadow: 0rem 0rem 1rem 0rem rgba(0,0,0,0.10);
  cursor: pointer;
  outline: none;
  margin-top: 0.25rem;
  padding-top: 1rem;
  padding-bottom: 1rem;
  min-width: inherit;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  position: absolute;
  z-index: 999;
}

.emotion-4 >*:not(:last-child) {
  margin-bottom: 0.5rem;
}

.emotion-3 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  height: 40px;
  width: auto;
  padding-left: 0.75rem;
  padding-right: 0.5rem;
  color: #FFFFFF;
  background: hsla(222, 63%, 43%, 1);
}

.emotion-3:disabled {
  background: hsla(0, 0%, 90%, 1);
  color: hsla(0, 0%, 45%, 1);
}

.emotion-3:hover {
  color: #FFFFFF;
  background: hsla(222, 63%, 34%, 1);
}

.emotion-3:active {
  color: #FFFFFF;
  background: hsla(222, 63%, 25%, 1);
}

.emotion-3:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-3:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(222, 64%, 84%, 1);
}

.emotion-3 svg {
  margin: 0.25rem;
}

<div
    class="emotion-5"
  >
    <span
      style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 1px;"
    >
      <label
        for="downshift-1-toggle-button"
        id="downshift-1-label"
      >
        Date
      </label>
    </span>
    <button
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-labelledby="downshift-1-label downshift-1-toggle-button"
      class="emotion-3"
      id="downshift-1-toggle-button"
      type="button"
    >
      <div
        class="emotion-2"
      >
        <div
          class="emotion-1"
        >
          <svg
            class="emotion-0"
            fill="currentColor"
            height="1.25rem"
            viewBox="0 0 256 256"
            width="1.25rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="16rem"
              width="16rem"
            />
            <polyline
              fill="none"
              points="208 96 128 176 48 96"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="18"
            />
          </svg>
           7 days ago
        </div>
      </div>
    </button>
    <div
      aria-labelledby="downshift-1-label"
      class="emotion-4"
      id="downshift-1-menu"
      role="listbox"
      tabindex="-1"
    />
  </div>
  .emotion-5 {
  position: relative;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  margin: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
}

.emotion-1 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  visibility: visible;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.emotion-0 {
  width: 1.25rem;
  height: 1.25rem;
  min-height: 1.25rem;
  min-width: 1.25rem;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: unset;
  -webkit-box-align: unset;
  -ms-flex-align: unset;
  align-items: unset;
  -webkit-box-pack: unset;
  -ms-flex-pack: unset;
  -webkit-justify-content: unset;
  justify-content: unset;
  visibility: hidden;
  background: #FFFFFF;
  border: 1px solid hsla(0, 0%, 87%, 1);
  border-radius: 4px;
  box-shadow: 0rem 0rem 1rem 0rem rgba(0,0,0,0.10);
  cursor: pointer;
  outline: none;
  margin-top: 0.25rem;
  padding-top: 1rem;
  padding-bottom: 1rem;
  min-width: inherit;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  position: absolute;
  z-index: 999;
}

.emotion-4 >*:not(:last-child) {
  margin-bottom: 0.5rem;
}

.emotion-3 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  height: 40px;
  width: auto;
  padding-left: 0.75rem;
  padding-right: 0.5rem;
  color: #FFFFFF;
  background: hsla(222, 63%, 43%, 1);
}

.emotion-3:disabled {
  background: hsla(0, 0%, 90%, 1);
  color: hsla(0, 0%, 45%, 1);
}

.emotion-3:hover {
  color: #FFFFFF;
  background: hsla(222, 63%, 34%, 1);
}

.emotion-3:active {
  color: #FFFFFF;
  background: hsla(222, 63%, 25%, 1);
}

.emotion-3:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-3:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(222, 64%, 84%, 1);
}

.emotion-3 svg {
  margin: 0.25rem;
}

<div
    class="emotion-5"
  >
    <span
      style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 1px;"
    >
      <label
        for="downshift-2-toggle-button"
        id="downshift-2-label"
      >
        Date
      </label>
    </span>
    <button
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-labelledby="downshift-2-label downshift-2-toggle-button"
      class="emotion-3"
      id="downshift-2-toggle-button"
      type="button"
    >
      <div
        class="emotion-2"
      >
        <div
          class="emotion-1"
        >
          <svg
            class="emotion-0"
            fill="currentColor"
            height="1.25rem"
            viewBox="0 0 256 256"
            width="1.25rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="16rem"
              width="16rem"
            />
            <polyline
              fill="none"
              points="208 96 128 176 48 96"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="18"
            />
          </svg>
           7 days ago
        </div>
      </div>
    </button>
    <div
      aria-labelledby="downshift-2-label"
      class="emotion-4"
      id="downshift-2-menu"
      role="listbox"
      tabindex="-1"
    />
  </div>
</DocumentFragment>
`;
